NGF: Update all existing guides to use cert-manager to manage secrets (#1933)
nginx/documentation: content/ngf/traffic-management/https-termination.md

## Configure HTTPS termination and routing

> For HTTPS, we need a certificate and key stored in a Secret. This Secret will live in a separate namespace, so we will need a ReferenceGrant in order to access it. We use cert-manager to issue the certificate from a local self-signed CA, which creates the `cafe-secret` Secret automatically.

Reviewer: ```suggestion
For HTTPS, the deployment requires a certificate and a private key stored in a Secret. The Secret lives in a separate namespace, so a ReferenceGrant is required to access it. cert-manager issues the certificate from a local self-signed CA and automatically creates the `cafe-secret` Secret.
```

To create the **certificate** namespace, copy and paste the following into your terminal:

```yaml
kubectl apply -f - <<EOF
apiVersion: v1
kind: Namespace
metadata:
  name: certificate
EOF
```

{{< include "ngf/deploy-cert-manager.md" >}}

{{< include "ngf/cert-manager-local-ca.md" >}}

> Create a `Certificate` for `cafe.example.com` in the `certificate` namespace. cert-manager will create the `cafe-secret` Secret containing `tls.crt`, `tls.key`, and `ca.crt`:

Reviewer: ```suggestion
Create a `Certificate` for `cafe.example.com` in the `certificate` namespace. cert-manager creates the `cafe-secret` Secret, which contains `tls.crt`, `tls.key`, and `ca.crt`.
```

...
```

> Now we will access the application over HTTPS. Since our certificate is signed by a local self-signed CA that curl does not trust, we will use curl's `--insecure` option to turn off certificate verification.

Reviewer: ```suggestion
Now access the application over HTTPS. Because the certificate is signed by a local self-signed CA that curl does not trust, use curl's `--insecure` option to turn off certificate verification.
```
